### PR TITLE
fix: changes to enroll syntax

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
- Modify enroll verb to accept a json for params. Added new class in at_commons EnrollParams.dart
- Rename the enrollApprovalId to enrollmentId for naming consistency
- Add "list" to "EnrollOperationEnum"
- Update the unit tests to for the new enroll verb syntax
- Export "enroll_params.dart" file

### CODE REVIEW

File: packages/at_commons/lib/src/verb/syntax.dart [hunk 1]
```diff
@@ -128,14 +128,10 @@ class VerbSyntax {
...
-      // Add colon only when operation is request, approve, deny or revoke. Do not add for list
-      r'(?:(?<=request|approve|deny|revoke):)?'
-      r'(?:(?:enrollmentId:(?<enrollmentId>[a-zA-Z0-9-]+))|'
-      r'(?:appName:(?<appName>[a-zA-Z0-9]+):)?'
-      r'(?:deviceName:(?<deviceName>[a-zA-Z0-9_]+):)?'
-      r'(?:namespaces:\[(?<namespaces>[a-zA-Z0-9;_,]+)\]:)?'
-      r'(?:totp:(?<totp>[0-9]+):)?'
-      r'(?:apkamPublicKey:(?<apkamPublicKey>.*)))?$';
+      // Positive lookbehind - Matches the value of operation named group
+      //  To ignore ":" if operation is list
```
Comment (gkc) [inline packages/at_commons/lib/src/verb/syntax.dart]: To be precise, this is actually to require ":" if operation is request|approve|deny|revoke

File: packages/at_commons/lib/src/verb/syntax.dart [hunk 2]
```diff
@@ -130,6 +130,8 @@ class VerbSyntax {
       r'^enroll:(?<operation>request|approve|deny|revoke|list)'
       // Positive lookbehind - Matches the value of operation named group
       //  To ignore ":" if operation is list
+      //  To be precise, this is actually to require ":"
+      //  if operation is request|approve|deny|revoke
```
Comment (gkc) [inline packages/at_commons/lib/src/verb/syntax.dart]: I didn't mean for you to literally transcribe what I wrote :-)

How about, `Look-behind to require ':' if operations is request|approve|deny|revoke`

On looking at this again, shouldn't we also require params when operation is request|approve|deny|revoke ?

An alternative regex would be to require `enroll:` then require either `list$` or `(request|approve|deny|revoke):(params fragment)$` which might be simpler to maintain

**Review by gkc — CHANGES_REQUESTED** [review]

File: packages/at_commons/lib/src/verb/syntax.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply packages/at_commons/lib/src/verb/syntax.dart]: Is the following regex looks good?
```^enroll:(?<operation>(list$)|(request|approve|deny|revoke|):)(?<enrollParams>.+)?$```

Comment (gkc) [reply packages/at_commons/lib/src/verb/syntax.dart]: I think you need to write some unit tests

Comment (sitaram-kalluri) [reply packages/at_commons/lib/src/verb/syntax.dart]: We have unit tests in "packages/at_commons/test/syntax_test.dart" to verify the enroll verb syntax. Will test the changes and add more tests to it.

**Review by gkc — APPROVED** [review]

**Pull request merged**